src/Entity/Product.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+final class Product
+{
+    private $id;
+    private $name;
+    private $price;
+    private $category;
+
+    public function __construct(string $name, float $price, $category)
+    {
+        $this->id = \App\ORM\Util\UUID::v4();
+        $this->name = $name;
+        $this->price = $price;
+        $this->category = $category ? $category : new Category('Burger');
```

Comment:
ok, usually I don't like constructors but I can do with it.
But.
If there is 1 thing you must avoid at any cost, is to put some logic in it.
Here setting a default category can make sense, but then creating/using a factory pattern would have been better.